lib/src/ast/selector/combinator.dart

```diff
@@ -0,0 +1,31 @@
+// Copyright 2022 Google Inc. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'package:meta/meta.dart';
+
+/// A combinator that defines the relationship between selectors in a
+/// [ComplexSelector].
+///
+/// {@category Selector}
+@sealed
+class Combinator {
```

Comment:
Not sure if it's worth doing in this PR, but would it make sense to use an [enhanced enum](https://dart.dev/guides/language/language-tour#declaring-enhanced-enums) for this? (they were added in Dart 2.17).